azure/build-and-deploy-azure.sh

```diff
@@ -0,0 +1,174 @@
+kubectl apply -f k8s/workspace-psa.yml
+
+
+# Install the secrets store CSI driver provider for Azure Key Vault
+helm repo add csi-secrets-store-provider-azure https://azure.github.io/secrets-store-csi-driver-provider-azure/charts
+
+helm list --namespace kube-system | grep 'csi-secrets-store' &>/dev/null
+if [ $? == 0 ]; then
+  echo "CSI driver is already installed"
+else
+  echo "Installing CSI driver"
+  helm install csi csi-secrets-store-provider-azure/csi-secrets-store-provider-azure --namespace kube-system
+fi
+
+#TO BE REPLACED WITH https://azure.github.io/azure-workload-identity/docs/installation.html
+echo "Add Azure pod identity to repo"
+helm repo add aad-pod-identity https://raw.githubusercontent.com/Azure/aad-pod-identity/master/charts
+
+helm list --namespace kube-system | grep 'aad-pod-identity' &>/dev/null
+if [ $? == 0 ]; then
+  echo "Azure pod identity chart already installed"
+else
+  helm upgrade --install aad-pod-identity aad-pod-identity/aad-pod-identity #NO LONGER WORKS BECAUSE OF OUR CONFIUGRATION (RESTRICTED IN DEFAULT)
+fi
+
+#END TO BE REPLACED WITH https://azure.github.io/azure-workload-identity/docs/installation.html
+
+
+# Preparing calico via Helm
+echo "preparing calico via Helm"
+helm repo add projectcalico https://docs.projectcalico.org/charts
+helm upgrade --install calico projectcalico/tigera-operator
+
+
+echo "Generate secret manager challenge secret 2"
+az keyvault secret set --name wrongsecret-2 --vault-name "${AZ_KEY_VAULT_NAME}" --value "$(openssl rand -base64 16)" >/dev/null
+
+echo "Generate secret manager challenge secret 3"
+az keyvault secret set --name wrongsecret-3 --vault-name "${AZ_KEY_VAULT_NAME}" --value "$(openssl rand -base64 16)" >/dev/null
+
+echo "Fill-out the secret volume manifest template"
+envsubst <./k8s/secret-volume.yml.tpl >./k8s/secret-volume.yml
+
+echo "Apply secretsmanager storage volume"
+kubectl apply -f./k8s/secret-volume.yml
+
+envsubst <./k8s/pod-id.yml.tpl >./k8s/pod-id.yml
+envsubst <./k8s/secret-challenge-vault-deployment.yml.tpl >./k8s/secret-challenge-vault-deployment.yml
+
+kubectl apply -f./k8s/pod-id.yml
+
+while [[ $(kubectl --namespace=default get pods -l "app.kubernetes.io/component=mic" -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True True" ]]; do echo "waiting for component=mic" && sleep 2; done
+while [[ $(kubectl --namespace=default get pods -l "app.kubernetes.io/component=nmi" -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True True True" ]]; do echo "waiting for component=nmi" && sleep 2; done
+
+
+echo "Installing metrics api-server"
+kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml
+
+wait
+
+# if passed as arguments, use those
+# otherwise, create new default values
+
+if [[ -z $APP_PASSWORD ]]; then
+  echo "No app password passed, creating a new one"
+  APP_PASSWORD="$( uuidgen | sed 's/[-]//g')"
+else
+  echo "App password already set"
+fi
+
```

Comment:
please output the app-password so people can easily find it back.

Reply:
It was actually viewed at the end but maybe adding an extra explanation here helps. No problem